app/helpers/application_helper.rb

```diff
@@ -325,27 +325,47 @@ def link_to_remove_section(linktext, form, class_of_section_to_remove="removeme"
     link_to_function(linktext, "remove_section(this, \"#{class_of_section_to_remove}\")", class: "hidden showme")
   end
 
-  def time_in_zone(time, zone=nil, user=User.current_user)
+  # show time in the time zone specified by the first argument
+  # add the user's time when specified in preferences
+  def time_in_zone(time, zone = nil, user = User.current_user)
     return ts("(no time specified)") if time.blank?
-    zone = ((user && user.is_a?(User) && user.preference.time_zone) ? user.preference.time_zone : Time.zone.name) unless zone
+ 
+    zone ||= (user&.is_a?(User) && user&.preference&.time_zone) ? user.preference.time_zone : Time.zone.name
     time_in_zone = time.in_time_zone(zone)
-    time_in_zone_string = time_in_zone.strftime('<abbr class="day" title="%A">%a</abbr> <span class="date">%d</span>
-                                                 <abbr class="month" title="%B">%b</abbr> <span class="year">%Y</span>
-                                                 <span class="time">%I:%M%p</span>').html_safe +
-                                          " <abbr class=\"timezone\" title=\"#{zone}\">#{time_in_zone.zone}</abbr> ".html_safe
-
-    user_time_string = "".html_safe
-    if user.is_a?(User) && user.preference.time_zone
```

Comment:
I think the addition of "set timezone" here has been broken, so I tried to fix it here, although it's not part of the reported issue on Jira. I commented about it in the Jira issue.

My current thought is that it seems like something that needs to be fixed eventually, and if I change a function to solve one issue anyway, I could fix something else about the same function as well. (And it's a short function.)

If reviewers don't like it, I'm happy to remove the additional part. I'd suggest leaving relevants tests as pending, and creating a new Jira issue about it, though.